vowpalwabbit/example.h

```diff
@@ -123,3 +172,12 @@ namespace VW
 {
 void return_multiple_example(vw& all, v_array<example*>& examples);
 }  // namespace VW
+std::string features_to_string(const example& ec);
```

Comment:
Some of these yyy_to_string() functions need access to different parts example.  Seems more clean to do it this way, rather than add additional parameters.

Reply:
I still don't see why intentionally restricting where these functions can be used is worth it? They should probably be implemented near the types anyway.

After looking at them, I can see some of then *do* require example, but definitely not all.